tracking-library-integration.md

```diff
@@ -0,0 +1,448 @@
+### Installing the JS library
+You can retrieve the tracking library code from the following URL:
+```
+https://tracking.metrilo.com/tracking.js?token=<YOUR_PROJECT_TOKEN>
+```
+Consider the `<YOUR_PROJECT_TOKEN>` parameter - here you should put your project's API token, which you could find on the Settings page of your project in Metrilo.
+
+After that place the received javascript snippet (below) right before closing the `<head>` tag.
+
+Example:
+```
+<script type="text/javascript">
+!function(){var p={};... /*code, code, code*/ ...window.metrilo.setUp("https://tracking.metrilo.com", "<YOUR_PROJECT_TOKEN>", [], false);
+</script>
+```
+:warning: **You should add the javascript snippet on every page you want to track!**
+
+### Identify a user
+Identifies current visitor explicitly by email. The common use cases where identification could be done are:
+
+* After order, because each order contains customer details
+* After subscription for some resource on your web-site
+* After login
+
+```javascript
+  var visitorEmail = 'johnnybravo@gmail.com'; // (required) Visitor's email address.
+  metrilo.identify(visitorEmail);
+```
+**Responses**  
+
+| Code | Message |  
+| --- | --- |  
+| 204 | Success response. |  
+| 400 | Bad request from client. |  
+| 401 | Project token is invalid. |  
+| 402 | Project requires payment. |  
+| 403 | The IP that the request came from is ignored. |  
+| 500 | Error from server while processing request. |  
+| 502 | Error from server accepting request. |  
+
+---
+
+
+### Tracking events
+The tracking events which the tracking library supports are:
+* [search](#search)
+* [viewPage](#viewPage)
+* [viewArticle](#viewArticle)
+* [viewCategory](#viewCategory)
+* [viewProduct](#viewProduct)
+* [addToCart](#addToCart)
+* [removeFromCart](#removeFromCart)
+* [applyTags](#applyTags)
+* [customEvent](#customEvent)
+
+#### search ####
+
+Visitor has searched in the shop (this supports only a plain query string without further filtering). This event was not available in the old tracking library.
+```javascript
+  var query = 'a plant'; // (required) Search query string that the visitor entered in the search form.
+  var url = 'http://myshop.com/search/?s=a+plant'; // (required) The URL that the visitor landed on when submitting the search form.
+
+  metrilo.search(query, url);
+```
+**Responses**  
+
+| Code | Message |  
+| --- | --- |  
+| 204 | Success response. |  
+| 400 | Bad request from client. |  
+| 401 | Project token is invalid. |  
+| 402 | Project requires payment. |  
+| 403 | The IP that the request came from is ignored. |  
+| 500 | Error from server while processing request. |  
+| 502 | Error from server accepting request. |  
+
+---
+
+#### viewPage ####
+Visitor has visited a shop page.
+
+```javascript
+  var url = 'http://myshop.com/amazing-page' // (required) Absolute URL linking the the visited page.
+  var params = {
+    name: 'My Amazing page' // (optional) The name of the page in the shop.
```

Comment:
As I already said, you should simply explain that everything in `params` is optional (which you've done but it's further down, you should do it before you start listing events), the rest is obviously not so no need to list whether every single parameters is optional or not